monitor/data_monitoring.py

```diff
@@ -152,6 +169,21 @@ def generate_report(self) -> bool:
             self.execution_properties['report_end'] = True
             self.execution_properties['success'] = self.success
             return self.success
+    
+    def send_report(self) -> bool:
+        elementary_html_file_path = os.path.join(self.config.target_dir, 'elementary.html')
```

Comment:
I'd call here the generate_report. Also, generate_report could return back elementary_html_file_path (in addition to the success flag). This way we could add to the path also the report creation time (we already add it to the json but it will be good to have it in the report name). WDYT?